---
title: "20160912_Main_text_figures"
author: "SZ"
date: "September 12, 2016"
output: html_document
---

```{r setup, include=FALSE}
library("devtools")
library("easysorter")
# devtools::install_github("AndersenLab/linkagemapping")
#devtools::install_github("AndersenLab/easysorter")
library("linkagemapping")
# devtools::install_github("wilkelab/cowplot")
library("cowplot")
library("tidyr")
library("dplyr")
# library("Cairo")
#devtools::install_github("Andersenlab/cegwas")
library("cegwas")
library("Hmisc")
library("genetics")
library("knitr")
library("reshape2")
library("ggplot2")
library("compute.es")
library("boot")
library("lme4")
library("data.table")
library("png")
library("grid")

dir.create("cache/")
dir.create("figures/")

output <- opts_knit$get("rmarkdown.pandoc.to")
opts_chunk$set(warning=F, message=F, echo=F)
opts_chunk$set(warning=F, message=F, echo=F)
opts_chunk$set(cache.path="cache/")
opts_chunk$set(fig.path="figures/")

output <- "html"
if (output=="html") opts_chunk$set(fig.width=12, fig.height=6)
```

# Load functions

```{r functions, cache=T}
source("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Figures/Plot_functions.R")
kinship_correction <- function(y, kin = cegwas::kinship){
  
  K <- kin[colnames(kin) %in% y$strain, rownames(kin) %in% y$strain]
  y <- y %>% dplyr::filter(strain %in% colnames(K))
  
  model = regress::regress(as.vector(y$value)~1,~K, pos = c(TRUE, TRUE))	
  
  #This err.cov is the same as err.cov in Dan's code using estVC
  err.cov = regress::summary.regress(model)$sigma[1]*K+regress::summary.regress(model)$sigma[2]*diag(nrow(K))
  
  eW = eigen(err.cov, symmetric = TRUE)
  
  if(min(eW$values) < 0 && abs(min(eW$values)) > sqrt(.Machine$double.eps)){
  }else{
    eW$values[eW$values <= 0] = Inf
  } 
  
  err.cov = eW$vector %*% diag(eW$values^-0.5) %*% t(eW$vector)
  new.pheno <- err.cov %*% y$value
  
  new.pheno <- data.frame(strain = y$strain, corrected_pheno = new.pheno)
  
  return(new.pheno)
}


quick_es <- function(df, trt, cond, parentals, comp_strains){
  
  # a d of 1, we know that the two groups' means differ by one standard deviation; a d of .5 tells us that the two groups' means differ by half a standard deviation
  
  stat_df <- df %>%
    dplyr::filter(trait == trt, condition==cond, strain %in% parentals)%>%
    dplyr::select(strain, phenotype)%>%
    dplyr::group_by(strain) %>%
    dplyr::summarise(mean_p = mean(phenotype),
              sd_p = sd(phenotype),
              counts = n())%>%
    data.frame()
  
  output <- mes(stat_df[1,2], stat_df[2,2], stat_df[1,3], stat_df[2,3], stat_df[1,4], stat_df[2,4],
                level = 95, cer = 0.2, dig = 2, verbose = FALSE, id=NULL, data=NULL)
  
  stat_df <- df %>%
    dplyr::filter(trait == trt, condition==cond, strain %in% comp_strains)%>%
    dplyr::select(strain, phenotype)%>%
    dplyr::group_by(strain) %>%
    dplyr::summarise(mean_p = mean(phenotype),
              sd_p = sd(phenotype),
              counts = n())%>%
    data.frame()
  
  output1 <- mes(stat_df[1,2], stat_df[2,2], stat_df[1,3], stat_df[2,3], stat_df[1,4], stat_df[2,4],
                level = 95, cer = 0.2, dig = 2, verbose = FALSE, id=NULL, data=NULL)
  
  eff <- output1[1,4]/output[1,4]
  
  return(eff)
}
vcor <- function (df, quantile_cutoff_high = 0.9, quantile_cutoff_low = 0.1, 
          variant_severity = c("MODERATE", "SEVERE"), gene_types = "ALL", 
          kin = kinship, condition_trait) 
{
  intervals <- df %>% 
    na.omit() %>% dplyr::distinct(CHROM, startPOS, endPOS, trait, .keep_all = TRUE) %>% 
    dplyr::distinct(CHROM, startPOS, trait, .keep_all = TRUE) %>% 
    dplyr::distinct(CHROM, endPOS, trait, .keep_all = TRUE) %>% dplyr::arrange(CHROM, startPOS)
  if (condition_trait == T) {
    intervals <- df %>% na.omit() %>% 
      tidyr::separate(trait,into = c("condition", "trait"), sep = "_") %>% 
      dplyr::distinct(CHROM, startPOS, endPOS, condition, .keep_all = TRUE) %>% 
      dplyr::distinct(CHROM, startPOS, condition, .keep_all = TRUE) %>% 
      dplyr::distinct(CHROM, endPOS, condition, .keep_all = TRUE) %>% 
      dplyr::arrange(CHROM, startPOS) %>% tidyr::unite(trait, 
                                                       condition, trait, sep = "_")
  }
  strains <- as.character(na.omit(unique(df$strain)))
  intervalGENES <- list()
  for (i in 1:nrow(intervals)) {
    print(paste(100 * signif(i/nrow(intervals), 3), "%", sep = ""))
    
    nstrains <- data.frame(df) %>% 
      na.omit() %>% 
      dplyr::filter(trait == as.character(intervals[i, "trait"]))
    nstrains <- length(unique(nstrains$strain))
    chr <- as.character(intervals[i, ]$CHROM)
    left <- intervals[i, ]$startPOS
    right <- intervals[i, ]$endPOS
    region_of_interest <- paste0(chr, ":", left, "-", right)
    snpeff_output <- snpeff(region = region_of_interest, 
                            severity = variant_severity, elements = gene_types)
    pruned_snpeff_output <- snpeff_output %>% 
      dplyr::filter(strain %in% strains) %>% 
      dplyr::filter(!is.na(impact)) %>% 
      dplyr::distinct(CHROM,  POS, strain, effect, gene_id, .keep_all = TRUE) %>% 
      dplyr::arrange(effect) %>% 
      dplyr::select(CHROM, POS,  REF, ALT, GT, effect, nt_change, aa_change, gene_name, gene_id, transcript_biotype, feature_type, strain) %>% 
      dplyr::group_by(CHROM, POS, effect) %>% 
      dplyr::filter(!is.na(GT), GT != "HET") %>% 
      dplyr::mutate(num_allele = ifelse(GT == "REF", 0, ifelse(GT == "ALT", 1, NA))) %>% 
      dplyr::mutate(num_alt_allele = sum(num_allele,na.rm = T), num_strains = n()) %>% 
      dplyr::filter(num_alt_allele/num_strains > 0.05) %>% 
      dplyr::filter(num_strains > nstrains * 0.8) %>% 
      dplyr::ungroup()
    if (nrow(pruned_snpeff_output) > 0) {
      interval_df <- df %>% 
        dplyr::filter(CHROM == chr, startPOS == left, endPOS == right) %>% 
        dplyr::group_by(trait, CHROM, startPOS, endPOS) %>% dplyr::filter(log10p == max(log10p)) %>% 
        dplyr::distinct(trait, startPOS,  endPOS, peakPOS, strain, .keep_all = TRUE) %>% 
        dplyr::ungroup() %>% 
        dplyr::select(trait, startPOS, endPOS, peakPOS, strain, log10p, pheno_value = value) %>% 
        na.omit()
      correct_it <- list()
      for (j in 1:length(unique(interval_df$trait))) {
        temp_pheno <- dplyr::filter(interval_df, trait == unique(interval_df$trait)[j]) %>% 
          dplyr::select(trait, strain, value = pheno_value)
        correct_it[[j]] <- kinship_correction(temp_pheno) %>% 
          dplyr::mutate(trait = unique(interval_df$trait)[j])
      }
      correct_df <- dplyr::rbind_all(correct_it)
      pheno_snpeff_df <- pruned_snpeff_output %>% 
        dplyr::left_join(., interval_df, by = "strain", copy = TRUE) %>% 
        dplyr::distinct(strain, trait, pheno_value, gene_id, CHROM, POS, aa_change, .keep_all = TRUE) %>% 
        dplyr::group_by(trait, CHROM, POS, effect, feature_type) %>% 
        dplyr::left_join(., correct_df, by = c("strain", "trait")) %>% 
        dplyr::group_by(trait, CHROM, POS, effect, feature_type) %>% 
        dplyr::mutate(number_strains = n())%>%
        dplyr::filter(number_strains > nstrains * 0.5) %>%
        dplyr::mutate(corrected_spearman_cor = cor.test(corrected_pheno, num_allele, method = "spearman", use = "pairwise.complete.obs")$p.value, 
                      spearman_cor = cor.test(pheno_value, num_allele, method = "spearman", use = "pairwise.complete.obs")$p.value) %>% 
        dplyr::ungroup() %>% 
        dplyr::mutate(abs_spearman_cor = abs(corrected_spearman_cor)) %>% 
        dplyr::filter(abs_spearman_cor < quantile(abs_spearman_cor, probs = quantile_cutoff_low, na.rm = T)) %>% 
        dplyr::ungroup() %>% 
        dplyr::arrange(abs_spearman_cor)
      intervalGENES[[i]] <- pheno_snpeff_df
    }
    else {
      intervalGENES[[i]] <- NA
    }
  }
  return(intervalGENES)
}
```


# Figure 1 -  mapping

```{r linkage data, cache=T}
genos <- data.frame(data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Ancillary/N2xCB4856_598RIAILs_gen.csv",
                                      header = T))
load("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Ancillary/marker_pos_conversion.Rda")

temp_genos <- do.call(cbind, lapply(genos[,4:ncol(genos)], function(x){
  temp <- data.frame(gen = x)
  temp <- dplyr::mutate(temp, num = ifelse(gen =="AA",1,
                                           ifelse(gen=="AB",0,NA)))
  temp <- dplyr::select(temp, -gen)
}))

riail_genotypes <- data.frame(genos[,1:3], temp_genos)
colnames(riail_genotypes) <- c("id","chr","CM",seq(1,ncol(riail_genotypes)-3))

riail_genotypes <- left_join(riail_genotypes, conv, by = "id")

link.pheno <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/RIAIL_phenotypes_etop_amsa.csv")

link.pheno.table <- link.pheno%>%
  dplyr::filter((trait == "median.TOF"& condition == "etoposide")|(trait == "f.L1"&condition == "amsacrine"))%>%
  dplyr::select(condition, strain, trait, phenotype)

write.table(link.pheno.table, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS2_LM-phenotypes.csv",
            quote=FALSE, sep=",", row.names = F)

link.maps <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/RIAIL_annotatedLODs_etop_medianTOF.csv")

write.table(link.maps, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS3_LM-Etoposide-medianTOF.csv",
            quote=FALSE, sep=",", row.names = F)

```


```{r alt figure1}

etoposide <- dplyr::filter(data.frame(link.maps), trait == "etoposide.median.TOF")
etop_link_pl <- maxlodplot1(data.frame(etoposide))+
  ggplot2::theme(axis.text.x = ggplot2::element_text(size = 0, face = "bold", color = "black"), 
                 axis.text.y = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                 axis.title.x = ggplot2::element_text(size = 0,face = "bold", color = "black", vjust = -0.3),
                 axis.title.y = ggplot2::element_text(size = 10, face = "bold", color = "black", vjust = 2),
                 strip.text.x = ggplot2::element_text(size = 10,  face = "bold", color = "black"),
                 strip.text.y = ggplot2::element_text(size = 10, face = "bold", color = "black"), 
                 plot.title = ggplot2::element_text(size = 0,  face = "bold", vjust = 1), 
                 legend.position = "none",  
                 panel.background = ggplot2::element_rect(color = "black",  size = 1),
                 strip.background = ggplot2::element_rect(color = "black",size = 1),
                 plot.background = element_blank()) 


gwas.pheno <-  data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/GWAS_plot_phenotypes_amsa_etop_tenip_xk.csv")[,1:3]

write.table(gwas.pheno, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS4_GWAS-phenotypes.csv",
            quote=FALSE, sep=",", row.names = F)


gwas.maps <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/GWAS_processed_maps_tenip_amsa_etop_xk.csv")

gwas.maps.etop.amsa <- data.frame(gwas.maps)%>%
  dplyr::filter(grepl("etopo|amsa",trait))

write.table(gwas.maps.etop.amsa, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS5_GWAS-mappings.csv",
            quote=FALSE, sep=",", row.names = F)

gwas.maps.etop <- dplyr::filter(gwas.maps.etop.amsa, grepl("etop",trait))

genes <- cegwas::process_correlations(variant_correlation(gwas.maps.etop,condition_trait = F))

genes_table <- dplyr::select(genes, CHROM,POS,REF,ALT,nt_change, aa_change, gene_name, gene_id,molecular_name, strain, GT, trait, pheno_value, spearman_pvalue=corrected_spearman_cor_p)

write.table(genes_table, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/PLoS-Genetics_Submission/20170605_Resubmission/final_changes/Tables/S6_Table.csv",
            quote=FALSE, sep=",", row.names = F)


manplot_e_gwas<- data.frame(gwas.maps) %>%
  dplyr::filter(!grepl("Mt",marker), log10p !=0)%>%
  dplyr::filter(trait %in% c("etoposide_median.tof")) %>%
  group_by(trait)%>%
  mutate(newBF = -log10(.05/n()))%>% # redfinfe BF and threshold to be at BF for plot
  mutate(newabove = ifelse(log10p > newBF, 1, 0))%>%
  dplyr::select(-BF,-aboveBF)%>%
  dplyr::rename(BF=newBF,aboveBF=newabove)%>%
  manplota(bf_line_color = 'gray50',point_size=0.5)


pxg_df <- data.frame(gwas.maps) %>%
  dplyr::filter(!grepl("Mt",marker), log10p !=0, grepl("^I_|^II_|X_", marker))%>%
  dplyr::filter(trait %in% c("etoposide_median.tof")) %>%
  pxg_plot1( loc = "II:11877529", color_strains = c("N2", "CB4856","N2", "CB4856"))

pxg_e_gwas<-pxg_df[[1]]+
      ggplot2::labs(y = "Animal\nLength", x = "Genotype at II:11877529")+
      ggplot2::theme(axis.text.x = ggplot2::element_text(size = 10, face = "bold", color = "black"), 
                 axis.text.y = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                 axis.title.x = ggplot2::element_text(size = 10,face = "bold", color = "black", vjust = -0.3),
                 axis.title.y = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                 strip.text.x = ggplot2::element_text(size = 10,  face = "bold", color = "black"),
                 strip.text.y = ggplot2::element_text(size = 10, face = "bold", color = "black"), 
                 plot.title = ggplot2::element_text(size = 0,  face = "bold", vjust = 1), 
                 legend.position = "none",  panel.background = ggplot2::element_rect(color = "black",  size = 1.2),
                 strip.background = ggplot2::element_rect(color = "black",size = 1.2),
                 plot.background = element_blank()) 
pxg_e_gwas


fine_pl<- genes %>% 
  dplyr::filter(strain == "CB4856")%>%
  ggplot(.)+
  aes(x = POS/1e6, y = -log10(corrected_spearman_cor_p), fill = GT)+
  geom_point(shape=21, size =1, alpha = .7)+
  scale_fill_manual(values = c("ALT" = "blue", "REF" = "orange"))+
  scale_color_manual(values = c("ALT" = "blue", "REF" = "orange"))+
  ggplot2::theme_bw() + 
  ggplot2::ylim(c(0,11))+
  ggplot2::theme(axis.text.x = ggplot2::element_text(size = 10, face = "bold", color = "black"),
                 axis.text.y = ggplot2::element_text(size = 10, face = "bold", color = "black"), 
                 axis.title.x = ggplot2::element_text(size = 10, face = "bold", color = "black", vjust = -0.3), 
                 axis.title.y = ggplot2::element_text(size = 8, face = "bold", color = "black"), 
                 strip.text.x = ggplot2::element_text(size = 10, face = "bold", color = "black"), 
                 strip.text.y = ggplot2::element_text(size = 10,  face = "bold", color = "black"), 
                 plot.title = ggplot2::element_text(size = 0,  face = "bold", vjust = 1),
                 panel.background = ggplot2::element_rect(color = "black",  size = 1.2), 
                 strip.background = ggplot2::element_rect(color = "black", size = 1.2),
                 legend.position = 'none')+
  labs(x = "Genomic Position (Mb)", y = expression(bold(-log[10](bolditalic(p)))))


cowplot::ggdraw() +
  draw_plot(etop_link_pl+ theme(panel.border=element_blank()), 0.014, .67, 1-.008, .33)+
  draw_plot(manplot_e_gwas[[1]]+scale_y_continuous(breaks = c(2,4,6,8,10),limits = c(0,10))+labs(x = "Genomic Position (Mb)", y = expression(bold(-log[10](p))))+
              theme(plot.title = ggplot2::element_text(size = 0,  face = "bold", vjust = 1),
                    axis.text.x = ggplot2::element_text(size = 0,  face = "bold", vjust = 1),
                    axis.text.y = ggplot2::element_text(size = 10,  face = "bold", vjust = 1),
                    axis.title.y = ggplot2::element_text(size = 10,  face = "bold", color = "black", vjust = -.6),
                    axis.title.x = ggplot2::element_text(size = 10,  face = "bold", color = "black", vjust = 2),
                    strip.background = element_blank(),
                    strip.text.x = element_blank()), 0, .345, 1, .36)+
  draw_plot(pxg_e_gwas + theme(panel.border=element_blank()), 0, 0, .33, .35)+
  draw_plot(fine_pl+ theme(panel.border=element_blank()), .34, 0, .65, .33)+
  draw_plot_label(c("A", "B","C","D"), c(0, 0,0,.33), c(1, .73,.4,.4), size = 14)

cowplot::ggsave(filename = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/PLoS-Genetics_Submission/20170605_Resubmission/final_changes/Figure1a.pdf",
                plot = last_plot(),
                width = 7,
                height = 3.75)

```


```{r interval_details}
# qtl2<-interval_summary("II:11671955-11908334")
# View(qtl2[[11]])
# 
# snp2<- snpeff("II:11671955-11908334",severity = c("ALL"))%>%
#   # data.frame()%>%
#   dplyr::filter(strain=="CB4856",GT=="ALT", effect != "intergenic_region", FT=="PASS")%>%
#   dplyr::distinct(gene_id,.keep_all = T)%>%
#   dplyr::distinct(CHROM,POS,.keep_all = T)
# 
# snp2_mh<- snpeff("II:11671955-11908334")%>%
#   # data.frame()%>%
#   dplyr::filter(strain=="CB4856",GT=="ALT", effect != "intergenic_region", FT=="PASS")%>%
#   dplyr::distinct(gene_id,.keep_all = T)%>%
#   dplyr::distinct(CHROM,POS,.keep_all = T)

```


# Figure 2 - Quantitative Compementation + swap

```{r}
quantcomp <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/top2_quant_comp_regressed.csv")

comp_table <- quantcomp%>%
  dplyr::filter(trait == "median.TOF")%>%
  dplyr::mutate(strain1 = factor(strain,levels = c("N2",
                            "CB4856" ,   
                            "N2/∆N2",
                            "N2/∆CB4856",
                            "CB4856/∆N2",
                            "CB4856/∆CB4856"),
                 labels = c("Bristol","Hawaii" ,   "Bristol/∆Bristol","Bristol/∆Hawaii","Hawaii/∆Bristol","Hawaii/∆Hawaii")))%>%
  dplyr::select(condition, strain=strain1, trait, phenotype)

write.table(comp_table, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS11_Reciprocal-Hemizygosity.csv",
            quote=FALSE, sep=",", row.names = F)

```


```{r}

quantcomp <- dplyr::filter(quantcomp, ! strain %in% c("N2/∆N2","CB4856/∆CB4856" ))

quantcomp.pl <- quantcomp %>%
  dplyr::filter(trait == "median.TOF")%>%
  ggplot(.)+
  aes(x = factor(strain,
                 levels = c("N2",
                            "CB4856" ,   
                            "N2/∆N2",
                            "N2/∆CB4856",
                            "CB4856/∆N2",
                            "CB4856/∆CB4856"),
                 labels = c("N2",
                            "CB4856" ,   
                            "N2/∆N2",
                            "N2/∆CB4856",
                            "CB4856/∆N2",
                            "CB4856/∆CB4856"),
                 ordered = T), 
      color = strain, 
      fill =  factor(strain,
                     levels = c("N2",
                                "CB4856" ,   
                                "N2/∆N2",
                                "N2/∆CB4856",
                                "CB4856/∆N2",
                                "CB4856/∆CB4856"),
                     labels = c("N2",
                                "CB4856" ,   
                                "N2/∆N2",
                                "N2/∆CB4856",
                                "CB4856/∆N2",
                                "CB4856/∆CB4856"),
                     ordered = T), y = phenotype)+
    geom_jitter(width=.25, size =1 , alpha =.7)+
  geom_boxplot(outlier.colour = NA, alpha=.7)+
  scale_color_manual(values = c("black", "black","black", "black", "black", "black"))+
  scale_fill_manual(values = c( "orange", "blue","gray50", "gray50","gray50", "gray50" ))+
  # scale_x_discrete(labels = c(expression(bold("N2")),
  #                             expression(bold("CB4856")),
  #                             expression(bold("N2 / N2")~bolditalic("\u0394top-2")),
  #                             expression(bold("N2 / CB4856")~bolditalic("\u0394top-2")),
  #                             expression(bold("CB4856 / N2")~bolditalic("\u0394top-2")),
  #                             expression(bold("CB4856 / CB4856")~bolditalic("\u0394top-2"))))+
  theme_bw()+
  theme(axis.text.x = element_text(size=0, face="bold", color="black"),
        axis.text.y = element_text(size=14, face="bold", color="black"),
        axis.title.x = element_text(size=0, face="bold", color="black", vjust=-.3),
        axis.title.y = element_text(size=16, face="bold", color="black"),
        strip.text.x = element_text(size=0, face="bold", color="black"),
        strip.text.y = element_text(size=0, face="bold", color="black"),
        plot.title = element_text(size=0, face="bold", vjust = 1),
        legend.position="none",
        panel.background = element_rect( color="black",size=1.2),
        strip.background = element_rect(color = "black", size = 1.2))+
  labs(y = "Animal Length", title = "Etoposide")

swap_theme <-   theme(axis.text.x = element_text(size=14, face="bold", color="black"),
        axis.text.y = element_text(size=14, face="bold", color="black"),
        axis.title.x = element_text(size=0, face="bold", color="black", vjust=-.3),
        axis.title.y = element_text(size=0, face="bold", color="black"),
        strip.text.x = element_text(size=0, face="bold", color="black"),
        strip.text.y = element_text(size=0, face="bold", color="black"),
        plot.title = element_text(size=0, face="bold", vjust = 1),
        legend.position="none",
        panel.background = element_rect( color="black",size=1.2),
        strip.background = element_rect(color = "black", size = 1.2),
        panel.border = element_rect( colour = "black"))


quantcomp.st <- quantcomp %>%
  dplyr::filter(trait == "median.TOF")

aov_res <- aov(quantcomp.st$phenotype ~ quantcomp.st$strain)
summary(aov_res)
TukeyHSD(aov_res)


regressed <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/CRISPR_swap_regress_phenotype.csv")


swap_table <- regressed%>%
  dplyr::filter( (trait == "mean.TOF"& condition == "Etoposide")|
                   (trait == "q10.TOF"& condition == "Amsacrine")|
                   (trait == "norm.n"& condition == "Teniposide")|
                   (trait == "median.TOF"& condition == "XK469"),
                   strain!="ECA400")%>%
  dplyr::mutate(strain1 = factor(strain, levels = c("N2",      "ECA401", "ECA402" ,"CB4856","ECA547", "ECA548" ,"ECA549"),
                                 labels =c("Bristol",     "Bristol(M)ECA401", "Bristol(M)ECA402" , "Hawaii","Hawaii(Q)ECA547", "Hawaii(Q)ECA548" ,"Hawaii(Q)ECA549")))%>%
  dplyr::select(condition, strain = strain1, trait, phenotype)

dact.regressed <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/top2_crisprswap_dactinomycin_assayregressed.csv")


medianEXT <- dact.regressed %>%
  dplyr::filter(trait == "norm.n")%>%
  dplyr::mutate(strain1 = factor(strain, levels = c("N2",      "ECA401", "ECA402" ,"CB4856","ECA547", "ECA548" ,"ECA549"),
                                 labels =c("Bristol",    "Bristol(M)ECA401", "Bristol(M)ECA402" ,  "Hawaii","Hawaii(Q)ECA547", "Hawaii(Q)ECA548" ,"Hawaii(Q)ECA549")))%>%
  dplyr::select(condition, strain = strain1, trait, phenotype)
  
all.swap_table <- bind_rows(list(swap_table,medianEXT))

all.swap_table <- all.swap_table%>%
  dplyr::group_by(strain,condition)%>%
  dplyr::mutate(mph = mean(phenotype),
                sph = sd(phenotype))%>%
  dplyr::filter(phenotype <= 2*sph+mph,
                phenotype >= mph-2*sph)%>%
    dplyr::select(condition, strain, trait, phenotype)

write.table(all.swap_table, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS12_Allele-swap-all-drugs.csv",
            quote=FALSE, sep=",", row.names = F)

# # # Prune based on bins
# bamfpruned <- bamf_prune(biopruned, drop = TRUE)%>%
#   tidyr::unite(well, condition,row,col,sep="",remove = F)

# Regress out the effect of assay first
# assayregressed <- regress(bamfpruned, assay = FALSE)

# keeping 402 and 547 because they looked the best
# assayregressed <- regress(dplyr::filter(biopruned, strain!="ECA400", strain != "ECA549", strain != "ECA401", strain != "ECA548"), assay = FALSE)

test <- regressed%>%
  dplyr::filter( trait == "mean.TOF", condition == "Etoposide",strain!="ECA400",strain!="ECA401",strain!="ECA547",strain!="ECA549")%>%
  group_by(strain)%>%
  mutate(med= median(phenotype) )%>%
  mutate(med1 = med+ 2*IQR(phenotype),
         allele = ifelse(strain %in% c("CB4856", "ECA400", "ECA401", "ECA402" ), "M797", "Q797"))%>%
    # dplyr::filter(trait == trit)%>%
  dplyr::group_by(strain)%>%
  dplyr::mutate(mph = mean(phenotype),
                sph = sd(phenotype))%>%
  dplyr::filter(phenotype <= 2*sph+mph,
                phenotype >= mph-2*sph)

etop_swap<-ggplot(test)+
  aes(x = factor(strain, levels = c("N2",  "ECA402", "CB4856","ECA548")), 
      y = phenotype, 
      fill = factor(strain, levels = c("N2",  "ECA402","CB4856",  "ECA548")))+
  scale_fill_manual(values=c("orange","gray50","blue","gray50","gray50","gray50","gray50"))+
  geom_jitter(width=.25, alpha = .7, size =1)+
  geom_boxplot(outlier.colour = NA, alpha=.7)+
  
  scale_x_discrete(labels = c("Bristol\n Q797","Bristol\n M797","Hawaii\n M797","Hawaii\n Q797"))+
  theme_bw()+
  swap_theme+
  labs( y = "Animal Length", title = "Etoposide")

swap.pl <- etop_swap



y1 <- .07
y2 <- .12

blck <- .05
spce <- .125
x1 <- .13
x2 <- .22
x3 <- .3125
x4 <- .407


ggdraw() +
  # # ADD PLOT
  draw_plot(quantcomp.pl, 0, .1, .5, .9) +
  draw_plot(swap.pl, .5, 0, .5, 1)+
  # #  LABELS
  # draw_label("Genome", x = 0.0, y = y1+.015,hjust = -.1, size = 12, fontface = 'bold')+
  # draw_label("Chr II", x = 0.0, y = y2+.018 ,hjust = -.16, size = 12, fontface = 'bold') +
  # # UPPER RECTANGLES
  geom_rect(aes(xmin = x1, xmax = x1+blck, ymin = y1, ymax = y1+y1*.007 +.02),
            colour = "black", fill = "orange", alpha =0.5)+
  geom_rect(aes(xmin = x2, xmax = x2+blck, ymin = y1, ymax = y1+y1*.007 +.02),
            colour = "black", fill = "blue", alpha =0.5)+
  geom_rect(aes(xmin = x3, xmax = x3+blck, ymin = y1, ymax = y1+y1*.007 +.02),
            colour = "black", fill = "orange", alpha =0.5)+
  geom_rect(aes(xmin = x4, xmax = x4+blck, ymin = y1, ymax = y1+y1*.007 +.02),
            colour = "black", fill = "blue", alpha =0.5)+
  # # LOWER RECTANGLES
  geom_rect(aes(xmin = x1, xmax = x1+blck, ymin = y2, ymax = y2+y2*.007 +.02),
            colour = "black", fill = "orange", alpha =0.5)+
  geom_rect(aes(xmin = x2, xmax = x2+blck, ymin = y2, ymax = y2+y2*.007 +.02),
            colour = "black", fill = "blue", alpha =0.5)+
  geom_rect(aes(xmin = x3, xmax = x3+blck, ymin = y2, ymax = y2+y2*.007 +.02),
            colour = "black", fill = "blue", alpha =0.5)+
  geom_rect(aes(xmin = x4, xmax = x4+blck, ymin = y2, ymax = y2+y2*.007 +.02),
            colour = "black", fill = "orange", alpha =0.5)+
  # # ADD TRIANGLES
  geom_point(aes(x = x3+.55*blck, y = y2 +.01225), shape = 24, fill = "gray", color = "black", size = 3)+
  geom_point(aes(x = x4+.55*blck, y = y2 +.01225), shape = 24, fill = "gray", color = "black", size = 3)+
  draw_plot_label(c("A", "B"), c(0,.49), c(1,1), size = 16)


ggsave("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/PLoS-Genetics_Submission/20170605_Resubmission/final_changes/Figure2_quantcomp_swap.pdf",
       height = 3.5,
       width = 7,units = "in",dpi = 300)
```


# Figure 4 - CRISPR SWAP other poisons

```{r}

swap_theme <-   theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        axis.text.y = element_text(size=12, face="bold", color="black"),
        axis.title.x = element_text(size=0, face="bold", color="black", vjust=-.3),
        axis.title.y = element_text(size=14, face="bold", color="black"),
        strip.text.x = element_text(size=12, face="bold", color="black"),
        strip.text.y = element_text(size=12, face="bold", color="black"),
        plot.title = element_text(size=16, face="bold", vjust = 1),
        legend.position="none",
        panel.background = element_rect( color="black",size=1.2),
        strip.background = element_rect(color = "black", size = 1.2),
        panel.border = element_rect( colour = "black"))

regressed <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/CRISPR_swap_regress_phenotype.csv")



# # # Prune based on bins
# bamfpruned <- bamf_prune(biopruned, drop = TRUE)%>%
#   tidyr::unite(well, condition,row,col,sep="",remove = F)

# Regress out the effect of assay first
# assayregressed <- regress(bamfpruned, assay = FALSE)

# keeping 402 and 547 because they looked the best
regressed <- dplyr::filter(regressed, strain!="ECA400", strain != "ECA549", strain != "ECA401", strain != "ECA548")

test <- regressed%>%
  dplyr::filter( trait == "norm.n", condition == "Teniposide")%>%
  group_by(strain)%>%
  mutate(med= median(phenotype) )%>%
  mutate(med1 = med+ 2*IQR(phenotype),
         allele = ifelse(strain %in% c("CB4856", "ECA400", "ECA401", "ECA402" ), "M797", "Q797"))%>%
    dplyr::group_by(strain)%>%
  dplyr::mutate(mph = mean(phenotype),
                sph = sd(phenotype))%>%
  dplyr::filter(phenotype <= 2*sph+mph,
                phenotype >= mph-2*sph)

aov_res <- aov(test$phenotype ~ test$strain)
summary(aov_res)
TukeyHSD(aov_res)

tenip_swap<-ggplot(test)+
  aes(x = factor(strain, levels = c("N2",  "ECA401", "ECA402","CB4856","ECA547", "ECA548")), 
      y = phenotype, 
      fill = factor(strain, levels = c("N2", "ECA401", "ECA402", "CB4856","ECA547", "ECA548")))+
  scale_fill_manual(values=c("orange","gray50","blue","gray50"))+
  geom_jitter(width=.25, size =1, alpha=.7)+
  geom_boxplot(outlier.colour = NA, alpha=.7)+
  facet_grid(condition~.)+
  scale_x_discrete(labels = c("Bristol\n Q797","Bristol\n M797","Hawaii\n M797", "Hawaii\n Q797"))+
  theme_bw()+
  swap_theme+
  labs( y = "Brood\nSize")+
  theme(axis.text.x = element_text(size=0, face="bold", color="black"),
        strip.background = element_rect(colour=NA, fill=NA))





test <- regressed%>%
  dplyr::filter( trait == "q10.TOF", condition == "Amsacrine")%>%
  group_by(strain)%>%
  mutate(med= median(phenotype) )%>%
  mutate(med1 = med+ 2*IQR(phenotype),
         allele = ifelse(strain %in% c("CB4856", "ECA400", "ECA401", "ECA402" ), "M797", "Q797"))%>%
    dplyr::group_by(strain)%>%
  dplyr::mutate(mph = mean(phenotype),
                sph = sd(phenotype))%>%
  dplyr::filter(phenotype <= 2*sph+mph,
                phenotype >= mph-2*sph)

aov_res <- aov(test$phenotype ~ test$strain)
summary(aov_res)
TukeyHSD(aov_res)


amsa_swap<-ggplot(test)+
  aes(x = factor(strain, levels = c("N2",  "ECA401", "ECA402","CB4856","ECA547", "ECA548")), 
      y = phenotype, 
      fill = factor(strain, levels = c("N2", "ECA401", "ECA402", "CB4856","ECA547", "ECA548")))+
  scale_fill_manual(values=c("orange","gray50","blue","gray50"))+
  geom_jitter(width=.25, size =1, alpha=.7)+
  geom_boxplot(outlier.colour = NA, alpha=.7)+
  scale_x_discrete(labels = c("Bristol\n Q797","Bristol\n M797","Hawaii\n M797", "Hawaii\n Q797"))+
    facet_grid(condition~.)+
  theme_bw()+
  swap_theme+
  labs(y = "Animal\nLength")+
   theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        strip.background = element_rect(colour=NA, fill=NA))

dact.regressed <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Raw_Data/top2_crisprswap_dactinomycin_assayregressed.csv")

trit <- "norm.n"
medianEXT <- dact.regressed %>%
  dplyr::filter(trait == trit)%>%
  dplyr::group_by(strain)%>%
  dplyr::mutate(mph = mean(phenotype),
                sph = sd(phenotype))%>%
  dplyr::filter(phenotype <= 2*sph+mph,
                phenotype >= mph-2*sph)

dact_swap<-ggplot(medianEXT) +
  aes(x = factor(strain,levels = c("N2","ECA401","ECA402","CB4856","ECA548","ECA549"),labels = c("N2","ECA401","ECA402","CB4856","ECA548","ECA549")), y = phenotype, fill=strain) +
  geom_jitter(width=.25, size =1, alpha=.7)+
  geom_boxplot(outlier.colour = NA, alpha = 0.7)+
  scale_fill_manual(values = c("N2" = "orange","CB4856" = "blue","ECA401" = "gray50","ECA402" = "gray50","ECA547" = "gray50","ECA549" = "gray50"))+
  theme_bw()+
    facet_grid(condition~.)+
  swap_theme+
  scale_x_discrete(labels = c("Bristol\n Q797","Bristol\n M797","Hawaii\n M797", "Hawaii\n Q797"))+
  labs( y = "Brood\nSize")+
  theme(axis.text.x = element_text(size=0, face="bold", color="black"),
        strip.background = element_rect(colour=NA, fill=NA))

aov_res <- aov(medianEXT$phenotype ~ medianEXT$strain)
summary(aov_res)
TukeyHSD(aov_res)

ggdraw() +
  draw_plot(tenip_swap, 0, .67, 1, .3) +
  draw_plot(dact_swap, 0, .38, 1, .3) +
  draw_plot(amsa_swap, 0, 0, 1, .38) +
  draw_plot_label(c("C", "B","A"), c(0,0,0), c(.425,.7,1), size = 16)


ggsave("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/PLoS-Genetics_Submission/20170605_Resubmission/final_changes/Figure4_CRISPRswap_poisons.pdf",
       height = 5,
       width = 4,
       dpi = 300)
```

# Figure 5 - Human edited cells

```{r}
reads <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/20170130_human-cell_reads.csv")

reads <- data.frame(t(reads))%>%
  dplyr::select(-X4)
colnames(reads) <- c("guide", "condition","amplicon","top2a_wt","top2a_edit","top2b_wt","top2b_edit")
reads <- reads[2:nrow(reads),]
reads$top2a_wt <- as.numeric(as.character(reads$top2a_wt))
reads$top2a_edit <- as.numeric(as.character(reads$top2a_edit))
reads$top2b_wt <- as.numeric(as.character(reads$top2b_wt))
reads$top2b_edit <- as.numeric(as.character(reads$top2b_edit))


# # # #  TOP2A
reads <- dplyr::mutate(reads, fraction_edited = ifelse(amplicon == "TOP2A", top2a_edit/ top2a_wt, top2b_edit/ top2b_wt))

top2a_o <- dplyr::filter(reads, guide == "No guide", 
                         # guide == "TOP2A", 
                         condition %in%c("Cellline Control D1"),
                         amplicon == "TOP2A")%>%  
  dplyr::select( condition, amplicon, fraction_edited)%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2a_o) <- c("amplicon","0")

top2a_ctrl <- dplyr::filter(reads, guide != "No guide", 
                            guide == "TOP2A", 
                            condition %in%c("Early Time Point", "No Drug"),
                            amplicon == "TOP2A")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2a_ctrl) <- c("guide","amplicon","1","2")

top2a_ctrl <- top2a_ctrl%>%
  dplyr::left_join(.,top2a_o,by="amplicon")

top2a_expt <- reads%>%
  dplyr::filter( guide == "TOP2A", 
                 !condition %in%c("Early Time Point"),
                 amplicon == "TOP2A")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  dplyr::left_join(.,top2a_ctrl, by = c("amplicon","guide"))%>%
  tidyr::gather(timepoint, base_fraction, -condition, -amplicon, -guide, -fraction_edited)%>%
  dplyr::mutate(fraction = ifelse(timepoint == "0", base_fraction,
                                  ifelse(timepoint == "1", base_fraction,
                                         ifelse(timepoint == "2", fraction_edited, NA))))%>%
  dplyr::mutate(fold = ifelse(timepoint == "0", base_fraction,
                              ifelse(timepoint == "1", base_fraction,
                                     ifelse(timepoint == "2", fraction_edited/base_fraction, NA))))%>%
  dplyr::mutate(l_frac = log10(fraction))

top2a_table <- left_join(reads, top2a_expt, by = c("condition", "amplicon","guide","fraction_edited"))%>%
  na.omit()

# # # #  TOP2b
top2b_o <- dplyr::filter(reads, guide == "No guide", 
                         # guide == "TOP2A", 
                         condition %in%c("Cellline Control D1"),
                         amplicon == "TOP2B")%>%  
  dplyr::select( condition, amplicon, fraction_edited)%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2b_o) <- c("amplicon","0")

top2b_ctrl <- dplyr::filter(reads, guide != "No guide", 
                            guide == "TOP2B", 
                            condition %in%c("Early Time Point", "No Drug"),
                            amplicon == "TOP2B")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  dplyr::group_by(guide, condition, amplicon)%>%
  dplyr::summarise(fraction_edited = mean(fraction_edited))%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2b_ctrl) <- c("guide","amplicon","1","2")

top2b_ctrl <- top2b_ctrl%>%
  dplyr::left_join(.,top2b_o,by="amplicon")

top2b_expt <- reads%>%
  dplyr::filter( guide == "TOP2B", 
                 !condition %in%c("Early Time Point"),
                 amplicon == "TOP2B")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  dplyr::group_by(guide, condition, amplicon)%>%
  dplyr::summarise(fraction_edited = mean(fraction_edited))%>%
  dplyr::left_join(.,top2b_ctrl, by = c("amplicon","guide"))%>%
  
  tidyr::gather(timepoint, base_fraction, -condition, -amplicon, -guide, -fraction_edited)%>%
  dplyr::mutate(fraction = ifelse(timepoint == "0", base_fraction,
                                  ifelse(timepoint == "1", base_fraction,
                                         ifelse(timepoint == "2", fraction_edited, NA))))%>%
  dplyr::mutate(fold = ifelse(timepoint == "0", base_fraction,
                              ifelse(timepoint == "1", base_fraction,
                                     ifelse(timepoint == "2", fraction_edited/base_fraction, NA))))%>%
  dplyr::mutate(l_frac = log10(fraction))


top2b_table <- left_join(reads, top2b_expt, by = c("condition", "amplicon","guide","fraction_edited"))%>%
  na.omit()

top2ab_table <- bind_rows(top2a_table,top2b_table)

write.table(top2ab_table, file = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Tables/TableS13_human_swap_reads.csv",
            quote=FALSE, sep=",", row.names = F)


```

```{r}
reads <- data.table::fread("~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/20170130_human-cell_reads.csv")

reads <- data.frame(t(reads))%>%
  dplyr::select(-X4)
colnames(reads) <- c("guide", "condition","amplicon","top2a_wt","top2a_edit","top2b_wt","top2b_edit")
reads <- reads[2:nrow(reads),]
reads$top2a_wt <- as.numeric(as.character(reads$top2a_wt))
reads$top2a_edit <- as.numeric(as.character(reads$top2a_edit))
reads$top2b_wt <- as.numeric(as.character(reads$top2b_wt))
reads$top2b_edit <- as.numeric(as.character(reads$top2b_edit))


# # # #  TOP2A
reads <- dplyr::mutate(reads, fraction_edited = ifelse(amplicon == "TOP2A", top2a_edit/ top2a_wt, top2b_edit/ top2b_wt))

top2a_o <- dplyr::filter(reads, guide == "No guide", 
                         # guide == "TOP2A", 
                         condition %in%c("Cellline Control D1"),
                         amplicon == "TOP2A")%>%  
  dplyr::select( condition, amplicon, fraction_edited)%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2a_o) <- c("amplicon","0")

top2a_ctrl <- dplyr::filter(reads, guide != "No guide", 
                            guide == "TOP2A", 
                            condition %in%c("Early Time Point", "No Drug"),
                            amplicon == "TOP2A")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  tidyr::spread(condition, fraction_edited)

colnames(top2a_ctrl) <- c("guide","amplicon","1","2")

top2a_ctrl <- top2a_ctrl%>%
  dplyr::left_join(.,top2a_o,by="amplicon")

top2a_expt <- reads%>%
  dplyr::filter( guide == "TOP2A", 
                 !condition %in%c("Early Time Point"),
                 amplicon == "TOP2A")%>%
  dplyr::select(guide, condition, amplicon, fraction_edited)%>%
  dplyr::left_join(.,top2a_ctrl, by = c("amplicon","guide"))%>%
  tidyr::gather(timepoint, base_fraction, -condition, -amplicon, -guide, -fraction_edited)%>%
  dplyr::mutate(fraction = ifelse(timepoint == "0", base_fraction,
                                  ifelse(timepoint == "1", base_fraction,
                                         ifelse(timepoint == "2", fraction_edited, NA))))%>%
  dplyr::mutate(fold = ifelse(timepoint == "0", base_fraction,
                              ifelse(timepoint == "1", base_fraction,
                                     ifelse(timepoint == "2", fraction_edited/base_fraction, NA))))%>%
  dplyr::mutate(l_frac = log10(fraction))

sw.pl <- top2a_expt%>%
  dplyr::filter(timepoint != "0", grepl("Mid|No|Actin", condition))%>%
  dplyr::mutate(xlabels = ifelse(timepoint == 1, "Pre-treatment", "Post-treatment"))%>%
  ggplot(.)+
  aes(x = factor(xlabels, levels = c("Pre-treatment", "Post-treatment"), 
                 labels = c("Pre-treatment", "Post-treatment"), 
                 ordered = T), 
      y = (fraction), group = condition, color = condition)+
  geom_line(size = 2)+
  scale_color_manual(values=c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#990000", "#D55E00", "#CC79A7"),
                     name="Experimental\nCondition",
                     breaks=c("No Drug","Amsacrine Hydrochloride Mid", "Actinomycin D Low", "Etoposide Mid","Teniposide Mid","XK469 Mid"),
                     labels=c("No Drug","Amsacrine", "Dactinomycin", "Etoposide",  "Teniposide", "XK469"))+
  scale_y_log10(limits=c(0.0001,0.1))+
  # xlim(0,.1)+
  theme_bw()+  
  theme(axis.text.x = element_text(size=14, face="bold", color="black"),
        axis.text.y = element_text(size=12, face="bold", color="black"),
        axis.title.x = element_text(size=0, face="bold", color="black", vjust=-.3),
        axis.title.y = element_text(size=14, face="bold", color="black"),
        strip.text.x = element_text(size=12, face="bold", color="black"),
        strip.text.y = element_text(size=12, face="bold", color="black"),
        plot.title = element_text(size=16, face="bold", vjust = 1),
        # legend.position="none",
        panel.background = element_rect( color="black",size=1.2),
        strip.background = element_rect(color = "black", size = 1.2),
        panel.border = element_rect( colour = "black"))+
  labs(y = "Fraction Edited Reads")


ggsave(sw.pl, filename = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Figures/Figure5_human_swaps.tiff",
       height = 4,
       width = 8)

ggsave(sw.pl, filename = "~/Dropbox/AndersenLab/LabFolders/Stefan/Manuscripts/20160105_Top2/Figures/Figure5_human_swaps.pdf",
       height = 4,
       width = 8)

```

